grading_algorithm.py

```diff
@@ -0,0 +1,246 @@
+
+    return error_count
+
+def word_position(expression, c_exp):
+    """
+    This is a function that determines if there is an error with word positions
+    Args:
+        expression (string): The student provided version of the expression
+        c_exp (string): The correct expression to be comapred to
+    
+    Returns:
+        error_count (int): the amount of times that a word order error occurs
+    """
+    l_expression = expression.lower()
+    expression_list = l_expression.split()
+    l_c_exp = c_exp.lower()
+    c_exp_list = l_c_exp.split()
+    error_count = 0
+    # Checks if the postiion of a word is the same in each expression
+    for x in range(len(expression_list)):
+        if x < len(c_exp_list):
+            if expression_list[x] != c_exp_list[x]:
+                error_count += 1
+
+    return error_count
+
+
+def word_count(expression, c_exp):
+    """
+    This is a function that determines if the number of words in the expressions are the same
+    Args:
+        expression (string): The student provided version of the expression
+        c_exp (string): The correct expression to be comapred to
+    
+    Returns:
+        correct_percent(int): The percentage the student got in this given section
+    """
+    l_expression = expression.lower()
+    expression_list = l_expression.split()
+    l_c_exp = c_exp.lower()
+    c_exp_list = l_c_exp.split()
+    error_count = 0
+    expression_len = len(expression_list)
+    c_exp_len = len(c_exp_list)
+    if expression_len == c_exp_len:
+        return 100
+    
+    if expression_len > c_exp_len:
+        error_count = expression_len - c_exp_len
+    else:
+        error_count = c_exp_len - expression_len
+
+    half_len = c_exp_len/2
+    correct_percent = 0
+    if error_count <= half_len:
+        correct_percent = (c_exp_len - error_count)/c_exp_len
+        correct_percent = correct_percent * 100
+        return correct_percent
+    else:
+        return correct_percent
+    
+def extra_words(expression, c_exp):
+    """
+    This is a function that determines if the student added extra words within their reformulation
+    Args:
+        expression (string): The student provided version of the expression
+        c_exp (string): The correct expression to be comapred to
+    
+    Returns:
+        correct_percent(int): The percentage the student got in this given sectione
+    """
+    l_expression = expression.lower()
+    expression_list = l_expression.split()
+    l_c_exp = c_exp.lower()
+    c_exp_list = l_c_exp.split()
+    error_count = 0
+
+    for x in expression_list:
+        if x in c_exp_list:
+            continue
+        else:
+            error_count += 1
+
+    half_len = len(c_exp_list)/2
+    correct_percent = 0
+    if error_count <= half_len:
+        correct_percent = (len(c_exp_list) - error_count)/len(c_exp_list)
+        correct_percent = correct_percent * 100
+        return correct_percent
+    else:
+        return correct_percent
+

```

Comment:
put this in if __name__ == "__main__"

Reply:
It looks like Dominic's comment was formatted by GitHub and had the underscores converted to bold. Here is what he means:
`
if __name__ == "__main__":
    c_exp = "Hello My Name is Elder Price."
    expression = "Hello My Pineapple is Elder Price."
    print(expression)
    print(c_exp)
    print("Your overall grade was:", grade_reformulation(expression, c_exp))
`
This prevents these from being run if this module is imported without removing these lines